sycl/source/detail/scheduler/commands.cpp

```diff
@@ -731,7 +736,7 @@ Command *Command::processDepEvent(EventImplPtr DepEvent, const DepDesc &Dep,
 
   ContextImplPtr DepEventContext = DepEvent->getContextImpl();
   // If contexts don't match we'll connect them using host task
-  if (DepEventContext != WorkerContext && !WorkerContext->is_host()) {
```

Comment:
wrong update, original condition DEC != WC && !WC->isHost() excludes case when both are on host. My condition checks only that WC is not host